Source/BugsnagThread.m

```diff
@@ -7,7 +7,103 @@
 //
 
 #import "BugsnagThread.h"
+#import "BugsnagCollections.h"
+#import "BugsnagStacktrace.h"
+#import "BugsnagKeys.h"
+
+@interface BugsnagStacktrace ()
+- (NSArray *)toArray;
+@end
+
+@interface BugsnagThread ()
+@property BugsnagStacktrace *trace;
+@end
+
+@interface BugsnagStacktrace ()
+@property NSMutableArray<BugsnagStackframe *> *trace;
+@end
 
 @implementation BugsnagThread
```

Comment:
I think this class would benefit from some method documentation.  The methods are well named but a single line summary of e.g. what enhancing a thread sets out to do would aid understanding.

Reply:
Added a few more comments in general - particularly around the thread enhancement.